src/Illuminate/Support/Number.php

```diff
@@ -28,9 +28,11 @@ class Number
      */
     public static function format(int|float $number, ?int $precision = null, ?int $maxPrecision = null, ?string $locale = null)
     {
+        $locale ??= static::checkLocale();
```

Comment:
Why this was moved to be an assignment instead of the inline check as before?

Reply:
Thanks for remembering the `value()` helper.

I believe it is better to do the assignment than to create the function to replace `static::$locale` and then do the inline check.